feature-libs/tms/adobe-launch/services/adobe-launch.service.ts

```diff
@@ -0,0 +1,77 @@
+import { Injectable, isDevMode, OnDestroy } from '@angular/core';
+import { CxEvent, EventService, WindowRef } from '@spartacus/core';
+import { TmsConfig } from '@spartacus/tms/core';
+import { merge, Subscription } from 'rxjs';
+
+export type AdobeLaunchDataLayer = (
+  data: any,
+  linkObject?: any,
+  eventObject?: any
+) => void;
+
+export interface AdobeLaunchWindow extends Window {
+  _trackData?: AdobeLaunchDataLayer;
+}
+
+/**
+ * This service interacts with the configured data layer object by pushing the Spartacus events to it.
+ */
+@Injectable({ providedIn: 'root' })
+export class AdobeLaunchService implements OnDestroy {
+  protected subscription = new Subscription();
+
+  constructor(
+    protected eventsService: EventService,
+    protected windowRef: WindowRef,
+    protected tmsConfig: TmsConfig
+  ) {}
+
+  /**
+   * This method is called only once to start collecting and dispatching events to Adobe Launch
+   */
+  collect(): void {
+    this.prepareDataLayer();
```

Comment:
Should we run this logic on a server side?

Reply:
see the comment below: https://github.com/SAP/spartacus/pull/10629#discussion_r565199515